pkg/codegen/jenny_schemareg.go

```diff
@@ -0,0 +1,88 @@
+package codegen
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing/fstest"
+
+	"github.com/grafana/codejen"
+	"github.com/grafana/grafana/pkg/cuectx"
+	"github.com/grafana/kindsys"
+	"github.com/grafana/thema"
+)
+
+// SchemaRegistryJenny generates lineage files into the "next" folder
+// of the local schema registry.
+func SchemaRegistryJenny(path string, latestRegistryDir string) OneToOne {
+	return &schemaregjenny{
+		path:              path,
+		latestRegistryDir: latestRegistryDir,
+	}
+}
+
+type schemaregjenny struct {
+	path              string
+	latestRegistryDir string
+}
+
+func (j *schemaregjenny) JennyName() string {
+	return "SchemaRegistryJenny"
+}
+
+func (j *schemaregjenny) Generate(kind kindsys.Kind) (*codejen.File, error) {
+	name := kind.Props().Common().MachineName
+	oldKindString, err := GetPublishedKind(name, "core", j.latestRegistryDir)
+	if err != nil {
+		return nil, err
+	}
+
+	var oldKind kindsys.Kind
+	if oldKindString != "" {
+		oldKind, err = loadCoreKind(name, oldKindString, "core")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// File is new - no need to compare with old lineage
+	if oldKind != nil {
+		// Check that maturity isn't downgraded
+		if kind.Maturity().Less(oldKind.Maturity()) {
+			return nil, fmt.Errorf("kind maturity can't be downgraded once a kind is published")
+		}
+
+		// Check that old schemas do not contain updates if maturity is greater than experimental
+		if kindsys.MaturityExperimental.Less(kind.Maturity()) && !thema.IsAppendOnly(oldKind.Lineage(), kind.Lineage()) {
+			return nil, fmt.Errorf("existing schemas in lineage %s cannot be modified", name)
+		}
+	}
+
+	core, ok := kind.(kindsys.Core)
+	if !ok {
+		return nil, fmt.Errorf("kind sent to SchemaRegistryJenny must be a core kind")
```

Comment:
```suggestion
		return nil, errors.New("kind sent to SchemaRegistryJenny must be a core kind")
```

Nit; I think if there's nothing to format/wrap, `errors.New` is the default convention.
However, looks like `SchemaRegistryJenny` could be a constant (as it is used above as well) and therefore the use of `fmt.Errorf` would be fine here in such case.

I leave it up to you, both are fine imho!